### ISSUE TITLE
Error responses are incorrectly validated against the normal response type

### ISSUE THREAD

**Issue by chriskuehl** [striglia/pyramid_swagger#121]
Hello!

When response validation is enabled, error responses are validated against the normal response type, often producing errors.

For example, I have an endpoint defined like:
```json
{
    "swaggerVersion": "1.2",
    "apiVersion": "1.0",
    "basePath": "http://localhost",

    "apis": [
        {
            "path": "/return_an_int",
            "operations": [
                {
                    "method": "GET",
                    "nickname": "return_an_int",
                    "parameters": [],
                    "type": "integer",
                    "responseMessages": [
                        {
                            "code": 404,
                            "message": "Could not find an int"
                        }
                    ]
                }
            ]
        }
    ]
}
```

If my endpoint returns a 404 error, pyramid_swagger intercepts it and I end up with a 500 instead:

```html
HTTP/1.1 500 Internal Server Error
Content-Type: text/html; charset=UTF-8

<html>
 <head>
  <title>500 Internal Server Error</title>
 </head>
 <body>
  <h1>500 Internal Server Error</h1>
  The server has either erred or is incapable of performing the requested operation.<br/><br/>
u'Could not find an int.' is not of type 'integer'

Failed validating u'type' in schema:
    {u'type': 'integer'}

On instance:
    u'Could not find an int.'


 </body>
</html>
```

I traced this to [tween.py#L450](https://github.com/striglia/pyramid_swagger/blob/74479b4812c7052f534930edd7915704afcc0c2d/pyramid_swagger/tween.py#L450), where the response body is turned into either text or JSON, then passed to the validator. By the time the validator sees the content, it will have no idea what the status code is, so will be unable to properly validate the response.

An easy solution might be to not validate error responses at all, but I believe Swagger actually allows to specify types for error messages as well, so perhaps a correct fix would take those into account.

Many thanks! (Tested in v1.5.0-rc2.)

**Comment by chriskuehl** [comment]
It seems that handling of errors is not super-well-defined in Swagger 1.2; see e.g. swagger-api/swagger-core#474, so trying to validate errors seems likely to be troublesome. Skipping validation for error responses, at least in Swagger 1.2, seems reasonable. Any thoughts?

**Comment by striglia** [comment]
Hi Chris, thanks for the clear writeup :)

Agreed that response validation is pretty unclear in 1.2. As a general note, I've found that the 2.0 spec is wildly improved in most areas.

This library in general is very much focused on primarily supporting Swagger2.0 as the main use case, but I can make a swagger12 branch for ongoing bugfixes or minor features. This change would, strictly speaking, alter the interface but I think it's valuable enough (and broken enough right now) that it makes sense to include.

I'll take the rest of the discussion to the PR, but thanks for reporting and pitching in.

**Comment by chriskuehl** [comment]
Closing, this was fixed by #123. Thanks again!
